ui/src/components/executions/FilePreview.vue

```diff
@@ -0,0 +1,66 @@
+<template>
+    <csv v-if="extensionToMonacoLang === 'csv'" :value="formattedContent" />
+    <img v-else-if="extensionToMonacoLang === 'image'" :src="formattedContent" alt="Image output preview">
+    <editor v-else :model-value="formattedContent" :lang="extensionToMonacoLang" format-on-start />
+</template>
+<script>
+    import Editor from "../inputs/Editor.vue";
+    import * as ion from "ion-js";
+    import Papa from "papaparse";
+    import Csv from "../Csv.vue";
+
+    export default {
+        components: {Csv, Editor},
+        props: {
+            content: {
+                type: String,
+                required: true
+            },
+            extension: {
+                type: String,
+                required: true
+            }
+        },
+        computed: {
+            extensionToMonacoLang() {
+                switch(this.extension) {
+                case "json":
+                    return "json";
+                case "yaml":
+                case "yml":
+                case "ion":
+                    // little hack to get ion colored with monaco
+                    return "yaml";
+                case "csv":
+                    return "csv";
+                case "jpg":
+                case "jpeg":
+                case "png":
+                case "svg":
+                    return "image";
+                default:
```

Comment:
I don't see how it can be a security risk as I just display character. The only difference is that if I don't know the extension, there will be no syntax highlighting.

Reply:
Well security issues can hide in multiple way.

Anyway, as I commented somewhere else, binary format may crash the browser, and some files may also be one long line that (avro, parquet, ...) so better only display what we are sure can be displayed.